### PR TITLE
[manifest] #5 add a script to compute the version of package

### PR DESCRIPTION
**Pull request by PengTian0**
Add a script to compute the version of package.

The proposal of manifest build in rackhd release  is [here](https://groups.google.com/forum/#!topic/rackhd/v3LpRuQJvQg).

This PR depends on: https://github.com/RackHD/on-tools/pull/25 ; https://github.com/RackHD/on-tools/pull/26; https://github.com/RackHD/on-tools/pull/28; https://github.com/RackHD/on-tools/pull/29

### CODE REVIEW

**Comment by PengTian0** [comment]
@RackHD/corecommitters

File: manifest-build-tools/application/version_generator.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, EMC, Inc.
+
+"""
+The script compute the version of a package, just like:
+1.1-1-devel-20160809150908-7396d91
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/version_generator.py
```
Comment (PengTian0) [inline manifest-build-tools/application/version_generator.py]: Yes. I will submit another commit to add more comment.

File: manifest-build-tools/application/version_generator.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    def __init__(self, repo_dir, manifest_repo_dir):
+        """
+        :return:
+ 
+        """
+        self._repo_dir = repo_dir
+        self._manifest_repo_dir = manifest_repo_dir
+        self.repo_operator = RepoOperator()
+ 
+    def generate_small_version(self):
```
Comment (panpan0000) [inline manifest-build-tools/application/version_generator.py]: can you please add more comment , to tell people what's small_version ,big_version and candidate_version?

File: manifest-build-tools/application/version_generator.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+        (out, err) = proc.communicate()
+        if proc.returncode == 0:
+            return out.strip()
+        else:
+            print "Failed to parse version in debian/changelog due to {0}".format(err)
+            return None
+
+        
+    def generate_candidate_version(self):
```
Comment (panpan0000) [inline manifest-build-tools/application/version_generator.py]: how about call it generate_version_stage(self) ?
people may use the term "stage of development" for those words: alpha, beta, devel ,RC...
https://en.wikipedia.org/wiki/Software_release_life_cycle

File: manifest-build-tools/application/version_generator.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        """
+        generate the candidate version according to branch
+        return: devel ,if the branch is master
+                rc, if the branch is not master
+        """
+        current_branch = self.repo_operator.get_current_branch(self._repo_dir)
+        candidate_version = ""
```
Comment (panpan0000) [inline manifest-build-tools/application/version_generator.py]: same here.
by default, version_stage is empty, means it's official release.

File: manifest-build-tools/application/version_generator.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    parser.add_argument("--manifest-repo-dir",
+                        required=True,
+                        help="the directory of manifest repository",
+                        action="store")
+
+    parser.add_argument("--is-official-release",
+                        default=False,
+                        help="This release if official",
+                        action="store_true")
+    parser.add_argument('--parameter-file',
```
Comment (panpan0000) [inline manifest-build-tools/application/version_generator.py]: is this param used ?
btw "will used" --> "will be used"

File: manifest-build-tools/application/version_generator.py (hunk 2 above)
Comment (PengTian0) [inline manifest-build-tools/application/version_generator.py]: Sure!

File: manifest-build-tools/application/version_generator.py (hunk 3 above)
Comment (PengTian0) [inline manifest-build-tools/application/version_generator.py]: Okay!

File: manifest-build-tools/application/version_generator.py (hunk 5 above)
Comment (PengTian0) [inline manifest-build-tools/application/version_generator.py]: Thanks for your catching!
I will fix it.

**Comment by panpan0000** [comment]
:+1:

File: manifest-build-tools/application/version_generator.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+"""
+The script compute the version of a package, just like:
+1.1-1-devel-20160809150908-7396d91
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/version_generator.py 
+--repo-dir /home/onrack/rackhd/release/rackhd-repos/PengTian0/b/b/on-http
+--manifest-repo-dir /home/onrack/rackhd/release/rackhd-repos/PengTian0/b/build-manifests
+--is-official-release
```
Comment (changev) [inline manifest-build-tools/application/version_generator.py]: Minor suggestion:

I think this example could be more proper if ```--is-official-release true```
```--is-official-release``` with no ```true``` or ```false``` may not appear in actual application scenarios.

Comment (PengTian0) [inline manifest-build-tools/application/version_generator.py]: The parameter type is Boolean.
If command with ```--is-official-release```, that means the parameter is true.
If command without it, that means the parameter is false.

**Pull request merged**